tech-debt: Regex injection: `new RegExp(\`\\b${symbolName}\\b\`)` uses raw symbolName witho

## Description

Regex injection: `new RegExp(\`\\b${symbolName}\\b\`)` uses raw symbolName without escapeRegExp — symbols containing regex metacharacters (e.g. `$foo`, `arr[0]`) will cause malformed patterns or runtime errors; rxml/references-provider.ts correctly uses escapeRegExp

## Problem

Found at packages/pike-lsp-server/src/features/navigation/definition.ts:664,711. Auto-discovered by DGA Architect agent.

## Expected Behavior

See description above.

## Suggested Approach

Follow existing codebase patterns for this type of change.

## Affected Files

packages/pike-lsp-server/src/features/navigation/definition.ts:664,711

## Acceptance Criteria

- [ ] Code change implemented
- [ ] Tests added/updated
- [ ] TypeScript compiles: `bun run typecheck`

**Priority:** P1 | **Type:** tech-debt | Auto-discovered

**Compile failed:**

shell: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1
  stdout: error TS2688: Cannot find type definition file for 'node'.
  The file is in the program because:
    Entry point of type library 'node' specified in compilerOptions
packages/pike-lsp-server/tsconfig.json(3,3): error TS5107: Option 'moduleResolution=node10' is deprecated and will stop functioning in TypeScript 7.0. Specify compilerOption '"ignoreDeprecations": "6.0"' to silence this error.
  Visit https://aka.ms/ts6 for migration information.

  stderr: 
  Command failed: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1

**Compile failed:**

shell: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1
  stdout: packages/pike-lsp-server/src/features/rxml/references-provider.ts(428,1): error TS1128: Declaration or statement expected.

  stderr: 
  Command failed: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1

--- a/packages/pike-lsp-server/src/features/editing/completion-member-access.ts
+++ b/packages/pike-lsp-server/src/features/editing/completion-member-access.ts
@@ -345,9 +345,7 @@ function collectClassMembers(classSymbol: PikeSymbol, doc: DocumentCacheEntry): 
   for (const inheritChild of inheritChildren) {
     const parentClassName = getSymbolClassname(inheritChild) ?? inheritChild.name;
     if (parentClassName) {
-      const parentClass = doc.symbols.find(
-        s => s.kind === 'class' && s.name === parentClassName
-      );
+      const parentClass = doc.symbols.find(s => s.kind === 'class' && s.name === parentClassName);
       if (parentClass?.children) {
         for (const parentMember of parentClass.children) {
           if (parentMember.kind !== 'inherit' && parentMember.name) {

--- a/packages/pike-lsp-server/src/features/navigation/definition.ts
+++ b/packages/pike-lsp-server/src/features/navigation/definition.ts
@@ -17,6 +17,7 @@ import type { ExpressionInfo, PikeSymbol, InheritanceInfo } from '@pike-lsp/pike
 import { readFile } from 'node:fs/promises';
 import { handleDirectiveNavigation } from './definition-directives.js';
 import { getWordAtPositionGeneric } from '../utils/pike-identifier.js';
+import { escapeRegExp } from '../rxml/references-provider.js';
 
 const uriDecodeLog = new Logger('Navigation');
 
@@ -661,7 +662,7 @@ async function findSymbolTextInIncludedFiles(
     return null;
   }
 
-  const pattern = new RegExp(`\\b${symbolName}\\b`);
+  const pattern = new RegExp(`\\b${escapeRegExp(symbolName)}\\b`);
   for (const include of cached.dependencies.includes) {
     try {
       const content = await readFile(include.resolvedPath, 'utf-8');
@@ -708,7 +709,7 @@ async function findSymbolInDirectIncludes(
 
   const source = document.getText();
   const includeRegex = /^#include\s+["<]([^">]+)[">]/gm;
-  const symbolRegex = new RegExp(`\\b${symbolName}\\b`);
+  const symbolRegex = new RegExp(`\\b${escapeRegExp(symbolName)}\\b`);
   const currentFilePath = uriToFilePath(uri);
 
   let includeMatch = includeRegex.exec(source);

